### PR TITLE
Explicitly cancel blocked requests (attempt 2) (uplift to 1.16.x)

### PR DESCRIPTION
**Pull request by iefremov**
Uplift of #7030
Fixes https://github.com/brave/brave-browser/issues/10063

Approved, please ensure that before merging: 
- [x] You have checked CI and the builds, lint, and tests all pass or are not related to your PR. 
- [x] You have tested your change on Nightly. 
- [x] The PR milestones match the branch they are landing to. 

After you merge: 
- [x] The associated issue milestone is set to the smallest version that the changes is landed on.

### CODE REVIEW

**Review by antonok-edm — APPROVED** [review]
LGTM

**Review by kjozwiak — APPROVED** [review]
Uplift into `1.16.x` approved after deliberating with @brave/uplift-approvers. QA has also verified the PR on Nightly as per https://github.com/brave/brave-core/pull/7030#issuecomment-724777175. @pes10k also confirmed that the issue was resolved once updated to the latest Nightly.

**Pull request merged**
